ui/src/main/java/com/xmartlabs/bigbang/ui/common/recyclerview/BaseRecyclerViewAdapter.java

```diff
@@ -317,4 +318,24 @@ public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
   public int getItemViewType(int position) {
     return types.indexOf(items.get(position).getType());
   }
+
+  /** Removes all items and notifies that the data has changed. */
+  @MainThread
+  public void clearAll() {
+    items.clear();
+    notifyDataSetChanged();
+  }
+
+  /**
+   * Removes all items in the recyclerView of a specified type.
+   *
+   * @param itemType The specified type of items to be removed.
+   */
+  @MainThread
+  public void clearItems(@NonNull RecycleItemType itemType) {
+    Stream.ofNullable(items)
+        .filter(value -> value.getType().equals(itemType))
+        .distinct()
```

Comment:
what do you think of using `toList()` and then `removeItems`? It's an option, maybe this way is better.